### ISSUE TITLE
Rifts and Vehicle Combat

### ISSUE THREAD

**Issue by gravagehulk** [javierriveracastro/betteroll-swade#426]
Does Better Rolls 2 treat vehicle weapons differently? I've been playing round with Rifts, which has a lot of vehicles and robots and power armor. When you put a pilot in the seat of a Robot and add vehicle weapons, the attacks don't seem to resolve against targets like regular weapons do. Clicking the "roll" button on the card does nothing.  The only way I could get it to kind of work was to manually create a weapon on the pilot that simulates the stats of the vehicle.

**Comment by SalieriC** [comment]
BR2 currently doesn't support vehicle weapons I think. There is no proper way of determining the shooter yet (may or may not the the operator of the vehicle, that's something the system needs to address however and it was suggested multiple times. There currently are two main solutions: Put the weapons on the shooter or let the shooter roll Shooting and then roll the damage with the default system rolls from the vehicle.

**Comment by gravagehulk** [comment]
The thing is, I did test putting the weapons on the shooter. Not only does it encumber the shooter (yes, I know I can zero out the weight), it still doesn’t resolve against the target. It seems like vehicle weapons in general weren’t producing results, but I’ll test again.

**Comment by gravagehulk** [comment]
FYI: The creator of Swade Tools was able to get this working. I don't know exactly what he did, but I sent him some console errors and a json of one of the vehicle weapons and now the attacks resolve against the target and roll damage normally. So, it seems like this can be done.

This is a big part of Rifts. Vehicles are a huge part of the system, as there is power armor, robots, attack vehicles, etc.

**Comment by SalieriC** [comment]
I don't know how vehicle weapons are set up in RIFTS but there shouldn't be a problem applying their damage. 🤔 
Getting the Skill from the driver isn't the issue anyway, the issue is that the driver may not be the shooter. I don't know how that works in RIFTS, but thinking Ships for example...

**Comment by gravagehulk** [comment]
Well, In most of the vehicles, you can only have one “pilot”.  That would be the shooter. I don’t think you can even put more than one pilot in a vehicle. That’s assumed to be the person controlling all aspects of the vehicle.

**Comment by SalieriC** [comment]
Yes, but exactly that is the problem that needs attention in SWADE core. It makes little sense that the driver also operates all the weapon systems. But I think we're talking about different things here.

**Comment by gravagehulk** [comment]
Yeah, I think making Better Rolls2 work mechanically is unrelated. The shooter will always be the pilot. I don’t see SWADE core changing that any time soon.

**Comment by javierriveracastro** [comment]
It looks like there are some issues mixed here. If I got it correctly:

- RIFT vehicle weapons don't apply damage against vehicles. In the base system vehicular weapons are just the same as weapons, and seem to be working right. As i don't own RIFTS, can you make a json export of a weapon that doesn't work, post it here, and tell us exactly what is happening and what you are expecting to happen.

- Shooting from weapons equipped on the vehicle, assuming that the pilot is the shooter. Than for sure can be done, but it is a new feature, not a bug. So someday.

**Comment by gravagehulk** [comment]
Attached is a json of one of the vehicle weapons. When I make an attack from the weapon on the vehicle, NOTHING happens. No card, nothing. If I take the same vehicle weapon and put it on the pilot, it will attack. But you should be able to attack from the vehicles. Vehicles are kind of a big deal in Rifts. Lots of giant robots, power armor, hover cycles, etc. with weapons attached.
[fvtt-Item-heavy-rail-gun.json.zip](https://github.com/javierriveracastro/betteroll-swade/files/8372511/fvtt-Item-heavy-rail-gun.json.zip)

I also tried all of the other weapons on the vehicle. No card from Betterrolls. This is working with SWADE Tools after an update from the developer, but I'm not sure what he did.

![image](https://user-images.githubusercontent.com/69117052/160638279-5469631a-e51c-47db-be76-92fbc6f758af.png)

**Comment by javierriveracastro** [comment]
That's point 2, a feature request. I'm more worries about point 1, that could be a bug.

If the weapon is shot from a character and it does not apply damage to another vehicle it is a problem.

**Comment by gravagehulk** [comment]
Okay, no problem. I just wanted to report it. I can use SWADE Tools for my automation. It seems like no one is familiar with Rifts except me. Maybe it's just too new or people aren't interested. Thanks for taking a look.

**Comment by javierriveracastro** [comment]
This weapon seems to work when equipped on a character. At least on my test:

![imaxe](https://user-images.githubusercontent.com/8113700/160639190-14f59908-fc98-400d-895a-2aeb76d149d9.png)

But for that shaken thing in a vehicle... that is a bug.

**Comment by gravagehulk** [comment]
Yeah, it does, but that's not how it's supposed to work. First, you'd have to zero out the weight of the weapons or encumbrance will come into play. Second, it requires you to essentially open two character sheets to make one attack (as GM), so it's not ideal.

**Comment by javierriveracastro** [comment]
Yes. I agree. But is not a bug, so it will have less priority. Someday it will be done.

**Comment by gravagehulk** [comment]
okay, thanks.

**Comment by necron-ninety-nine** [comment]
Hey you have to the the eyeball on the weapon in the vehicle sheet
now it will bring up  the SWADE core card instead of the BR2 card now this is with legacy sheet as you are using i dont know about the new sheet

**Comment by javierriveracastro** [comment]
The new sheet has been introduced yesterday (4/2/25 d/m/Y), so it is quite new. I will give a couple of days to stabilize and probably wait until I can check that the SciFi companion doesn't modify it before starting to work on supporting it.

**Comment by ddbrown30** [comment]
I believe this should all be working in the latest version. Please open an new issue with updated information if you're still having problems.
